aws/processcreds/provider.go

```diff
@@ -0,0 +1,342 @@
+	SecretAccessKey string
+	SessionToken    string
+	Expiration      *time.Time
+}
+
+// retrieveFn executes the 'credential_process' and returns the credentials.
+func (p *Provider) retrieveFn(ctx context.Context) (aws.Credentials, error) {
+	out, err := p.executeCredentialProcess(ctx)
+	if err != nil {
+		return aws.Credentials{Source: ProviderName}, err
+	}
+
+	// Serialize and validate response
+	resp := &credentialProcessResponse{}
+	if err = json.Unmarshal(out, resp); err != nil {
+		return aws.Credentials{Source: ProviderName}, awserr.New(
+			ErrCodeProcessProviderParse,
+			fmt.Sprintf("%s: %s", errMsgProcessProviderParse, string(out)),
+			err)
+	}
+
+	if resp.Version != 1 {
+		return aws.Credentials{Source: ProviderName}, awserr.New(
+			ErrCodeProcessProviderVersion,
+			errMsgProcessProviderVersion,
+			nil)
+	}
+
+	if len(resp.AccessKeyID) == 0 {
+		return aws.Credentials{Source: ProviderName}, awserr.New(
+			ErrCodeProcessProviderRequired,
+			errMsgProcessProviderMissKey,
+			nil)
+	}
+
+	if len(resp.SecretAccessKey) == 0 {
+		return aws.Credentials{Source: ProviderName}, awserr.New(
+			ErrCodeProcessProviderRequired,
+			errMsgProcessProviderMissSecret,
+			nil)
+	}
+
+	creds := aws.Credentials{
+		Source:          ProviderName,
+		AccessKeyID:     resp.AccessKeyID,
+		SecretAccessKey: resp.SecretAccessKey,
+		SessionToken:    resp.SessionToken,
+	}
+
+	// Handle expiration
+	if resp.Expiration != nil {
+		creds.CanExpire = true
+		creds.Expires = (*resp.Expiration).Add(-p.options.ExpiryWindow)
+	}
+
+	return creds, nil
+}
+
+// prepareCommand prepares the command to be executed.
+func (p *Provider) prepareCommand(ctx context.Context) (context.Context, context.CancelFunc, error) {
+
+	var cmdArgs []string
+	if runtime.GOOS == "windows" {
+		cmdArgs = []string{"cmd.exe", "/C"}
+	} else {
+		cmdArgs = []string{"sh", "-c"}
+	}
+
+	if len(p.originalCommand) == 0 {
+		p.originalCommand = make([]string, len(p.command.Args))
+		copy(p.originalCommand, p.command.Args)
+
+		// check for empty command because it succeeds
+		if len(strings.TrimSpace(p.originalCommand[0])) < 1 {
+			return nil, nil, awserr.New(
+				ErrCodeProcessProviderExecution,
+				fmt.Sprintf(
+					"%s: %s",
+					errMsgProcessProviderPrepareCmd,
+					errMsgProcessProviderEmptyCmd),
+				nil)
+		}
+	}
+
+	timeoutCtx, cancelFunc := context.WithTimeout(ctx, p.options.Timeout)
+
+	cmdArgs = append(cmdArgs, p.originalCommand...)
+	p.command = exec.CommandContext(timeoutCtx, cmdArgs[0], cmdArgs[1:]...)
+	p.command.Env = os.Environ()
+
+	return timeoutCtx, cancelFunc, nil
+}
+
+// executeCredentialProcess starts the credential process on the OS and
+// returns the results or an error.
+func (p *Provider) executeCredentialProcess(ctx context.Context) ([]byte, error) {
+	ctx, cancelFunc, err := p.prepareCommand(ctx)
+	if err != nil {
+		return nil, err
+	}
+	defer cancelFunc()
+
+	output := bytes.NewBuffer(make([]byte, 0, DefaultBufSize))
```

Comment:
Do we want to export `DefaultBufSize` the value cannot be set? Since we're not using the limit reader anymore probably don't need to export this value.